samples/bluetooth/hci_vs_scan_req/src/main.c

```diff
@@ -0,0 +1,228 @@
+};
+
+static const struct bt_data scan_rsp_data[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA(BT_DATA_NAME_COMPLETE, scan_data, sizeof(scan_data)),
+};
+
+static const char *bt_addr_le_str(const bt_addr_le_t *addr)
+{
+	static char str[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, str, sizeof(str));
+
+	return str;
+}
+
+/* Allow different TX power per role/connection. It requires CONFIG_BT_CTLR_TX_PWR_DYNAMIC_CONTROL.
+ * same as in samples/bluetooth/hci_pwr_ctrl/src
+ */
+static void set_tx_power(uint8_t handle_type, uint16_t handle, int8_t tx_pwr_lvl)
+{
+	struct bt_hci_cp_vs_write_tx_power_level *cp;
+	struct bt_hci_rp_vs_write_tx_power_level *rp;
+	struct net_buf *buf, *rsp = NULL;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_VS_WRITE_TX_POWER_LEVEL, sizeof(*cp));
+	if (!buf) {
+		printk("Unable to allocate command buffer\n");
+		return;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = sys_cpu_to_le16(handle);
+	cp->handle_type = handle_type;
+	cp->tx_power_level = tx_pwr_lvl;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_VS_WRITE_TX_POWER_LEVEL,
+				   buf, &rsp);
+	if (err) {
+		uint8_t reason = rsp ?
+			((struct bt_hci_rp_vs_write_tx_power_level *)
+			  rsp->data)->status : 0;
+		printk("Set Tx power err: %d reason 0x%02x\n", err, reason);
+		return;
+	}
+
+	rp = (void *)rsp->data;
+	printk("Actual Tx Power: %d\n", rp->selected_tx_power);
+
+	net_buf_unref(rsp);
+}
+
+/* Bluetooth specification doesn't allow the scan request event with legacy advertisements.
+ * Ref: Bluetooth Core Specification v5.4, section 7.7.65.19 "LE Scan Request Received event" :
+ *      "This event shall only be generated if advertising was enabled using the
+ *       HCI_LE_Set_Extended_Advertising_Enable command."
+ * Added a Vendor Specific command to add this feature and save RAM.
+ */
+static void enable_legacy_adv_scan_request_event(bool enable)
+{
+	struct bt_hci_cp_vs_set_scan_req_reports *cp;
+	struct net_buf *buf;
+	int err;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_VS_SET_SCAN_REQ_REPORTS, sizeof(*cp));
+	if (!buf) {
+		printk("%s: Unable to allocate HCI command buffer\n", __func__);
+		return;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->enable = (uint8_t) enable;
+
+	err = bt_hci_cmd_send(BT_HCI_OP_VS_SET_SCAN_REQ_REPORTS, buf);
+	if (err) {
+		printk("Set legacy cb err: %d\n", err);
+		return;
+	}
+}
+
+static bool vs_scanned(struct net_buf_simple *buf)
+{
+	struct bt_hci_evt_vs_scan_req_rx *evt;
+	struct bt_hci_evt_vs *vs;
+
+	vs = net_buf_simple_pull_mem(buf, sizeof(*vs));
+	evt = (void *)buf->data;
+
+	printk("%s subevent 0x%02x peer %s rssi %d\n", __func__,
+	       vs->subevent, bt_addr_le_str(&evt->addr), evt->rssi);
+
+	return true;
+}
+
+static int start_advertising(void)
+{
+	int err;
+
+	err = bt_hci_register_vnd_evt_cb(vs_scanned);
+	if (err) {
+		printk("VS user callback register err %d\n", err);
+		return err;
+	}
+	enable_legacy_adv_scan_request_event(true);
```

Comment:
```suggestion
	}
	
	enable_legacy_adv_scan_request_event(true);
```

Suggest to keep an empty line after ending `}` (not a requirement)

Reply:
Done